File: Day-67_(0008)Atoi.py
def myAtoi(self, s: str) -> int:
        j= []
        sint=""
        sig=""
        k=0
        flag=True
        fd=True
        fs=True

        for i in s:
            j.append(i)
        for i in range(len(j)):
            if j[i]==' ' and fd==True and fs==True:
                continue
            # elif j[i]=='0' and fd==True :
            #     continue
            elif j[i]=='0'  :
                sint+="0"
                fd=False
            elif j[i]=="-" and flag==True and fd==True:
                flag=False
                fs=False
                sig+="-"
            elif j[i]=="+" and flag==True:
                flag=False
                sig+="+"
                fs=False
            elif j[i].isalpha() :
                break
            elif j[i].isnumeric():
                sint+=j[i]
                fd=False
            else:
                break
        if sig=="-" and sint!="":
            k= 0 - (int(sint))
            print(k)


        if (sint!="" and sig=="+" )or (sint!="" and sig==""):   
            k=int(sint)
            print(k)
        if k<=-(2**31):
            k=-(2**31) 
        elif k>=(2**31) :
            k=(2**31) -(1)
        print(sint," " ,k)
        if sint=="":
            return 0
        else:
            return int(k)
s ="1223"
num  = myAtoi(s)
print(num)

File: Day-67_(0008)Atoi.cpp
#include<bits/stdc++.h>
using namespace std;

int myAtoi(string s) {
         bool strt=false;
        int sig=1;
        int ans=0;
        int i=0;
        //get rid of leading whitespace
        while(s[i]==' '){  i++;}
        
        for(;i<s.length();i++){
            //accept first sign and indiacate the number has started
            if(!strt && (s[i]=='-' ||s[i]=='+')){
                if(s[i]=='-'){
                    sig=-1;
                }
                strt=true;
            }
            //accept number assume sign + and indiacte the number started
           else if(s[i]>='0' && s[i]<='9'){
                  strt=true;
               //check overflow
                 if(ans>INT_MAX/10 || (ans==INT_MAX/10)&&(s[i]-'0'>7)){
                     return (sig==1)?INT_MAX:INT_MIN ;
                 }
               
                ans = (ans*10)+ (s[i]-'0');
            }
         //if the number started and any value other than 0-9 digits encountered 
           else if((s[i]<'0' || s[i]>'9')){
                break; 
            }
           
        }
        return sig*ans;
    }

int main(){

string s ="  -1223";
int num  = myAtoi(s);
cout<<num<<endl;

return 0;
    }